### PR TITLE
Add BindFields function tool

### PR DESCRIPTION
**Pull request by ahmadmuzakki29**
I've been working with graphql-go lately and realize I'm pretty lazy to create object type for every struct I had. So I made a binder/wrapper function to easily bind all fields in my struct into graphql object.

So if you have this struct 
```go
type Person struct {
    Name    string  `graph:"name"`
    Address string  `graph:"address"`
}
```
you can create person type with 
```go
personType := graphql.NewObject(graphql.ObjectConfig{
        Name:   "Person",
        Fields: BindFields(Person{}),
    })
```

in file `util_test.go` I made an example with Person struct that looked like this:
```go
type Person struct {
	Human //embedded Fields
	Name    string   `json:"name"` // standard scalar type
	Home    Address  `json:"home"` // struct type
	Hobbies []string `json:"hobbies"` // slice of scalar type 
	Friends []Friend `json:"friends"` // slice of struct type
}

type Human struct {
	Alive  bool    `json:"alive"`
	Age    int     `json:"age"`
	Weight float64 `json:"weight"`
}

type Friend struct {
	Name    string `json:"name"`
	Address string `json:"address"`
}

type Address struct {
	Street string `json:"street"`
	City   string `json:"city"`
}
```

because this function will scan through empty struct so this function cannot accept recursive type. that would make a stack-overflow error. And one more utility function which is `BindArgs` that can bind the args depending on provided params.

### CODE REVIEW

File: util.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		return Boolean
+	case reflect.Slice:
+		return getGraphList(tipe)
+	}
+	return String
+}
+
+func getGraphList(tipe reflect.Type) *List {
+	switch tipe {
+	case reflect.TypeOf([]int{}):
```
Comment (bbuck) [inline util.go]: Not a fan of all the `reflect` usage here, you've already got a `reflect.Type`, you can use it. like here `case tipe.Kind() == reflect.Slice && tipe.Elem().Kind() == reflect.Int` and other tests `case tipe.Kind() == reflect.Slice && tipe.Elem().Kind() == reflect.Int8`, etc...

**Comment by bbuck** [comment]
I'm still not a fan of auto-binding nested structs, which is exactly the opposite of how I (and all the demos I've ever seen of how to use/configure GraphQL go. The point is to make sub records searchable in their own right. For example, you _should_ be able to bind recursively. If you have a `me` query endpoint that returns a `Person` type, and a `Person struct` has a field `Friends []Person` that should return a `[Person]` and GraphQL's system handle it perfectly. That is what it's designed to do (not just recursive, but nested/linked data in this manner).

**Comment by ahmadmuzakki29** [comment]
if you mean nested structs by `Person` struct is `Human` then `Human` struct is supposed to be on the same level. so the `Person` struct is like having all the fields that `Human` has.
```go
type Person struct{
	Alive  bool    `json:"alive"`
	Age    int     `json:"age"`
	Weight float64 `json:"weight"`
	Name string    `json:"name"`
       Address Address `json:"address"`
       // and so on ....
}
```
in that struct the sub record is address, so yes you can query `Address` in their own right. 
But if you take example like `Friends` with type of `Person` this function will loop forever because it will bind the `Person` recursively.
appreciated your input, but I'm interested to hear another people opinion about this approach.

**Comment by zeucxb** [comment]
Waiting

**Comment by ahmadmuzakki29** [comment]
@zeucxb I updated the PR and hope it can be accepted to merge, if it's not feel free to just copy util.go and place it in your package

**Comment by chris-ramon** [comment]
This util function is very awesome!, @ahmadmuzakki29 thanks a lot for taking the time.

I believe this is the initial function for more useful always optional time-savers functions, let's improve the recursive part and other improvements on new PR's!, thanks! :ship:

**Comment by ahmadmuzakki29** [comment]
Sure thanks!

**Comment by axetroy** [comment]
I got the error

```bash
"reflect: call of reflect.Value.NumField on ptr Value"
```

and here is my struct, can anyone help?

```go
type Post struct {
	ID        string       `json:"id"`
	UID       string       `json:"uid"`
	Cid       string       `json:"cid"`
	ShortId   string       `json:"shortId"`
	Title     string       `json:"title"`
	Summary   string       `json:"summary"`
	Content   string       `json:"content"`
	Gallery   []string     `json:"gallery"`
	Like      int32        `json:"like"`
	Comment   int32        `json:"comment"`
	Share     int32        `json:"share"`
	View      int32        `json:"view"`
	Tags      []string     `json:"tags"`
	IsShow    bool         `json:"isShow"`
	IsActive  bool         `json:"isActive"`
	IsTop     bool         `json:"isTop"`
	CreatedAt rpcbase.Time `json:"createdAt"`
	UpdatedAt rpcbase.Time `json:"updatedAt"`
}
```

**Comment by bbuck** [comment]
@axetroy How are you calling the tool? Also it might be better to create your own issue for this, as this issue is related to merging the tool in, not bugs and/or issues arising from using it.

**Comment by ahmadmuzakki** [comment]
@axetroy I suspect rpcbase.Time causing it. please create the fields manually for CreatedAt and UpdatedAt, and set it like this
```
CreatedAt rpcbase.Time `json:"-"`
UpdatedAt rpcbase.Time `json:"-"`
```

**Comment by limoli** [comment]
Same problem with a struct having custom fields `time.Time`.
I think that bindFields **doesn't handle** time types.

**Pull request merged**